app/src/main/java/org/mifos/mobile/ui/fragments/BeneficiaryAddOptionsFragment.kt

```diff
@@ -119,21 +115,9 @@ class BeneficiaryAddOptionsFragment : BaseFragment() {
                         Constants.PERMISSIONS_CAMERA_STATUS)
             }
             RequestAccessType.EXTERNAL_STORAGE_READ -> {
-                requestPermission((activity as BaseActivity?)!!,
-                        Manifest.permission.READ_EXTERNAL_STORAGE,
-                        Constants.PERMISSIONS_REQUEST_READ_EXTERNAL_STORAGE,
-                        resources.getString(R.string.dialog_message_storage_permission_denied_prompt),
-                        resources
-                                .getString(R.string.dialog_message_read_storage_permission_never_ask_again),
-                        Constants.PERMISSIONS_STORAGE_STATUS)
-            }
-            RequestAccessType.EXTERNAL_STORAGE_WRITE -> {
-                requestPermission((activity as BaseActivity?)!!,
-                        Manifest.permission.WRITE_EXTERNAL_STORAGE,
-                        Constants.PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE,
-                        resources.getString(R.string.dialog_message_storage_permission_denied_prompt),
-                        resources.getString(R.string.dialog_message_write_storage_permission_never_ask_again),
-                        Constants.PERMISSIONS_STORAGE_STATUS)
+                requestPermissions( arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE,
```

Comment:
Why did you remove the first parameter: activity?

Reply:
That is for custom text in permission dialog also it is written 2 times so calls dialog 2 times so I added code for multiple permission with default text.